### ISSUE TITLE
Support jsonc and json5

### ISSUE THREAD

**Issue by shian15810** [chris48s/v8r#124]
Currently, running `v8r` on jsonc files like `tsconfig.json` fails with not being able to parse the json properly

Newer `ajv` version supports jsonc and json5, would love to see these supports land in `v8r` as well

Probably related to https://github.com/chris48s/v8r/pull/123

**Comment by chris48s** [comment]
As of version 0.9.0 files with a `.json5` or `.jsonc` extension are parsed as JSON5.

This is the same behaviour that ajv-cli implements and will cover most cases but it probably doesn't cover your example because files with a `.json` extension are still assumed to be actual JSON (not unreasonably).

In order to work for that case I would also need to add a flag (or setting - see https://github.com/chris48s/v8r/issues/149 ) to allow users to manually specify/override the format of a file.

**Comment by chris48s** [comment]
I've just released v0.12.0 which adds a config file format.

One of the available settings is `customCatalog`. A custom catalog entry can include a `parser` key which overrides the default inferred from the file extension if specified. This means you could set:

```json
{
  "customCatalog": {
    "schemas": [
      {
        "name": "tsconfig.json",
        "fileMatch": [
          "tsconfig.json"
        ],
        "location": "https://json.schemastore.org/tsconfig.json",
        "parser": "json5"
      }
    ]
  }
}
```

in your config file to tell v8r to validate `tsconfig.json` against https://json.schemastore.org/tsconfig.json and parse it as json5 instead of json.
